src/main/kotlin/app/revanced/patches/reddit/testing/patch/TotalAdPostPatch.kt

```diff
@@ -0,0 +1,179 @@
+package app.revanced.patches.reddit.testing.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.extensions.removeInstruction
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprintResult
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.patches.reddit.testing.annotations.AdPostCompatibility
+import app.revanced.patches.reddit.testing.fingerprints.AdPostFingerprint
+import app.revanced.patches.reddit.testing.fingerprints.NewAdPostFingerprint
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.builder.MutableMethodImplementation
+import org.jf.dexlib2.dexbacked.reference.DexBackedFieldReference
+import org.jf.dexlib2.iface.instruction.formats.Instruction22c
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.immutable.ImmutableMethod
+import org.jf.dexlib2.immutable.ImmutableMethodParameter
+
+@Patch
+@Name("hide-promoted")
+@Description("Removes promoted posts in all feeds.")
+@AdPostCompatibility
+@Version("0.0.1")
+class TotalAdPostPatch : BytecodePatch(
+    listOf(AdPostFingerprint, NewAdPostFingerprint)
+) {
+    /**
+     * Adds the method filterChildren(List<Any>): List<Any>
+     * to filter away all ILinks which are labeled as promoted.
+     **/
+    private fun addFilterChildrenMethod(result: MethodFingerprintResult) {
+        val filteringMethod = ImmutableMethod(
+            result.method.definingClass,
+            "filterChildren",
+            listOf(ImmutableMethodParameter("Ljava/util/List;", null, "children")),
+            "Ljava/util/List;",
+            AccessFlags.PRIVATE or AccessFlags.FINAL,
+            null,
+            null,
+            MutableMethodImplementation(4)
+        ).toMutable().apply {
+            /*
+            // Can be summarized as the following
+            fun filterChildren(p1: Iterable<Any>): List<Any> {
+                val filteredList = mutableListOf<Any>()
+                val iterator = p1.iterator()
+                while (iterator.hasNext()) {
+                    val item = iterator.next()
+
+                    if (item is ILink) {
+                        if (!item.getPromoted()) {
+                            filteredList.add(item)
+                        }
+                    }
+                }
+                return filteredList
+            }
+            */
+            addInstructions(
+                0,
```

Comment:
Move it to integrations

Reply:
Using AbstractTransformInstructionsPatch?
I am not too sure what you mean, first time touching this codebase :pensive: 

The whole functionality of addFilterChildrenMethod or just the string?

Reply:
Basically, you should inject [as little code as possible](https://github.com/revanced/revanced-patches/blob/docs/docs/4_structure_and_conventions.md#-conventions) and allocate your code to [ReVanced Integrations](https://github.com/revanced/revanced-integrations). Integrations are compiled classes that are merged into the app. You can then inject calls to methods from these classes.

Reply:
Wow that's a nice feature, I will get working on it tomorrow :+1:

Reply:
Use the @RequiresIntegrations annotation on the patch. If it is present, ReVanced Patcher will merge integrations. Currently very WIP as there's some changes awaiting how integrations will work in the future.